cpp/include/core/concepts/GameRules.hpp

```diff
@@ -13,12 +13,22 @@ template <typename GR, typename GameTypes, typename GameResultsTensor, typename
 concept GameRules = requires(const State& const_state, const StateHistory& const_history,
                              State& state, StateHistory& history, group::element_t sym,
                              core::seat_index_t last_player, core::action_t last_action,
-                             GameResultsTensor& results) {
+                             GameResultsTensor& results, action_mode_t action_mode) {
+  // Initialize the state.
   { GR::init_state(state) };
+  // Get the legal moves for the state history.
   { GR::get_legal_moves(const_history) } -> std::same_as<typename GameTypes::ActionMask>;
+  // Get the action mode for the state.
```

Comment:
The first 3 comments here serve no purpose - they provide no info beyond what the function name already provides.